src/singletons.js

```diff
@@ -2,20 +2,25 @@ const { web3 } = require('./setup');
 const ether = require('./ether');
 const send = require('./send');
 
+const truffleContract = require('truffle-contract');
+
 const {
   ERC1820_REGISTRY_ABI,
   ERC1820_REGISTRY_ADDRESS,
   ERC1820_REGISTRY_DEPLOY_TX,
 } = require('./data');
 
-const registry = new web3.eth.Contract(ERC1820_REGISTRY_ABI, ERC1820_REGISTRY_ADDRESS);
+const web3ERC1820Registry = new web3.eth.Contract(ERC1820_REGISTRY_ABI, ERC1820_REGISTRY_ADDRESS);
+
+const truffleERC1820Registry = truffleContract({ abi: ERC1820_REGISTRY_ABI });
+truffleERC1820Registry.setProvider(web3.currentProvider);
```

Comment:
This will not work correctly. We have to grab the provider later, when we return the contract to the user, because of configurability.

Reply:
Ah, good catch, I forgot to change this when adding reconfigurability.